.travis.yml

```diff
@@ -25,38 +26,93 @@ jobs:
+      env: PHPCS_VERSION="2.2.0"
     - php: 5.6
+      env: PHPCS_VERSION="3.0.2" LINT=1
+    - php: 5.6
+      env: PHPCS_VERSION="2.4.0"
     - php: 5.5
       # As the latest Debian does not support PHP 5.5 anymore, we need to force using 'trusty'.
       dist: trusty
+      env: PHPCS_VERSION="dev-master" LINT=1
+    - php: 5.5
+      # As the latest Debian does not support PHP 5.5 anymore, we need to force using 'trusty'.
+      dist: trusty
+      env: PHPCS_VERSION="2.6.1"
     - php: 5.4
       # As the latest Debian does not support PHP 5.4 anymore, we need to force using 'trusty'.
       dist: trusty
+      env: PHPCS_VERSION="3.5.3" LINT=1
+    - php: 5.4
+      # As the latest Debian does not support PHP 5.4 anymore, we need to force using 'trusty'.
+      dist: trusty
+      env: PHPCS_VERSION="2.8.1"
     - php: 5.3
       # As the latest Debian does not support PHP 5.3 anymore, we need to force using 'precise'.
       dist: precise
-      script:
-        - find . -type f -name "*.php" -print0 | xargs -0 -n1 php -l
-        - travis_wait composer install --no-interaction --no-progress --no-scripts --no-suggest --optimize-autoloader --prefer-dist --verbose
-        - composer install-codestandards
-        - ./vendor/bin/phpcs -i
-        - ./vendor/bin/phpcs -s --standard=PHPCompatibility --runtime-set testVersion "${TRAVIS_PHP_VERSION:0:3}" --exclude=PHPCompatibility.Upgrade.LowPHP ./src/
+      env: PHPCS_VERSION="2.9.2"
+    - php: 5.3
+      # As the latest Debian does not support PHP 5.3 anymore, we need to force using 'precise'.
+      dist: precise
+      env: PHPCS_VERSION="2.0.0"
 
   fast_finish: true
 
 before_install:
   # Speed up build time by disabling Xdebug.
   phpenv config-rm xdebug.ini || echo 'No xdebug config.'
 
-script:
-  - find . -type f -name "*.php" -print0 | xargs -0 -n1 php -l
+install:
+  - composer require --no-update --no-suggest --no-scripts squizlabs/php_codesniffer:${PHPCS_VERSION}
+  - |
+    if [[ ${PHPCS_VERSION:0:3} < "2.2" ]]; then
+      # Install PHPCompatibility 7.x for PHPCS < 2.2.
+      composer require --no-update --no-suggest --no-scripts phpcompatibility/php-compatibility:"^7.0"
+    elif [[ ${PHPCS_VERSION:0:3} < "2.3" ]]; then
+      # Install PHPCompatibility 8.x for PHPCS 2.2 < 2.3.
+      composer require --no-update --no-suggest --no-scripts phpcompatibility/php-compatibility:"^8.0"
+    fi
   - travis_wait composer install --no-interaction --no-progress --no-scripts --no-suggest --optimize-autoloader --prefer-dist --verbose
+  - |
+    if [[ ${PHPCS_VERSION:0:3} < "2.2" ]]; then
+      # Rename the PHPCompatibility directory as PHPCompatibility 7.x wasn't fully compatible with Composer yet.
+      mv ./vendor/phpcompatibility/php-compatibility ./vendor/phpcompatibility/PHPCompatibility
+    fi
+
+script:
+  - if [[ "$LINT" == "1" ]]; then if find . -path ./vendor -prune -o -name "*.php" -exec php -l {} \; | grep "^[Parse error|Fatal error]"; then exit 1; fi; fi
```

Comment:
I'm wondering if, instead of  using
```
find . -path ./vendor -prune -o -name "*.php" -exec php -l {} \; 
``` 
it might be easier to just use
```
find ./src/ -name "*.php" -exec php -l {} \;
```
🤔

Reply:
@Potherca I understand the question. I didn't in the PR as it's _really_ easy to overlook something like that and forget to change it again when - for instance - a `tests/` directory would be added at a later time.

As it is now, the logic is consistent with the in/exclusion rules also used in the `phpcs.xml.dist` file